src/cargo/sources/git/utils.rs

```diff
@@ -1458,6 +1458,10 @@ fn github_fast_path(
         if let Some(local_object) = local_object {
             headers.append(&format!("If-None-Match: \"{}\"", local_object))?;
         }
+        if let Ok(token) = std::env::var("GITHUB_TOKEN") {
+            // Avoid API rate limits if possible.
+            headers.append(&format!("Authorization: token {token}"))?;
```

Comment:
According to [the GitHub REST API doc](https://docs.github.com/en/rest/authentication/authenticating-to-the-rest-api?apiVersion=2022-11-28#about-authentication), both `token` and `Bearer` scheme work. The [example for GitHub Actions](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#example-2-calling-the-rest-api) uses `Bearer`. Should we use that instead?

> Note: In most cases, you can use Authorization: Bearer or Authorization: token to pass a token. However, if you are passing a JSON web token (JWT), you must use Authorization: Bearer.

Reply:
I think it depends on which scenario we want to address. If this is for CI, then bearer seems more reasonable?

Reply:
Sure, went ahead and changed it.